drainer/config.go

```diff
@@ -349,3 +347,22 @@ func (cfg *Config) adjustConfig() error {
 
 	return nil
 }
+
+func validateAddr(addr string, strict bool) error {
+	urllis, err := url.Parse(addr)
+	if err != nil {
+		return errors.Annotatef(err, "failed to parse addr %v", addr)
+	}
+
+	var host string
+	if host, _, err = net.SplitHostPort(urllis.Host); err != nil {
+		return errors.Annotatef(err, "invalid host %v", urllis.Host)
+	}
+
+	if !util.IsValidateListenHost(host) {
+		if !(strict && host == "127.0.0.1") {
```

Comment:
if not strict mode , it would return error

if strict mode and host is `localhost` or `127.xx.xx.xx(not 127.0.0.1)`, it would return error

is this what your want? @suzaku